client/packages/system/src/IndicatorsDemographics/DetailView/IndicatorsDemographics.tsx

```diff
@@ -75,54 +77,50 @@ const IndicatorsDemographicsComponent = () => {
   };
 
   const setter = (patch: RecordPatch<Row>) => {
-    const updatedDraft = { ...draft };
     const percentage = !patch.percentage ? 0 : patch.percentage;
     const percentageChange = percentage != draft[patch.id]?.percentage;
 
     setIsDirty(true);
 
+    const existingRow = draft[patch.id];
+    if (!existingRow) return;
```

Comment:
this would be better to have above `setIsDirty`

Reply:
ah very true, thanks